naga/src/back/msl/keywords.rs

```diff
@@ -346,6 +346,7 @@ pub const RESERVED: &[&str] = &[
     super::writer::MODF_FUNCTION,
     super::writer::ABS_FUNCTION,
     super::writer::DIV_FUNCTION,
+    super::writer::DOT_FUNCTION,
```

Comment:
We somehow need to add all the possible variants of the dot function to this list, not the prefix. It looks like the list is `pub` but we don't use it directly, so maybe make the list private, put a comment here saying `DOT_FUNCTION` variants are added below, and add something in the `KeywordSet` construction for them.

The list added here should somehow be bound to the set of possibilities that `get_dot_wrapper_function_helper_name` can return (so there's no risk that WGSL adds a new type in the future that gets added in one place but not the other). Maybe iterate `ScalarSet::CONCRETE_INTEGER`?

Reply:
I extracted these functions to the `type_methods` module and updated the code to use them.

I also eliminated `Box::leak`. Leaking a handful of strings for a global data structure is not a problem, but I worry about the cost of reviewing/waiving false positives when using leak checkers.